iodata/formats/wfn.py

```diff
@@ -494,23 +494,39 @@ def _dump_helper_section(f: TextIO, data: NDArray, fmt: str, skip: int, step: in
 DEFAULT_WFN_TTL = "WFN auto-generated by IOData"
 
 
+def prepare_dump(data: IOData):
+    """Check the compatibility of the IOData object with the WFN format.
+
+    Parameters
+    ----------
+    data
+        The IOData instance to be checked.
+    """
+    if data.mo is None:
+        raise PrepareDumpError("The WFN format requires molecular orbitals")
+    if data.obasis is None:
+        raise PrepareDumpError("The WFN format requires an orbital basis set")
+    if data.mo.kind == "generalized":
+        raise PrepareDumpError("Cannot write WFN file with generalized orbitals.")
+    if data.mo.occs_aminusb is not None:
+        raise PrepareDumpError("Cannot write WFN file when mo.occs_aminusb is set.")
+    for shell in data.obasis.shells:
+        if any(kind != "c" for kind in shell.kinds):
+            raise PrepareDumpError("The WFN only supports Cartesian MolecularBasis.")
```

Comment:
When we try to dump a `wfn` with spherical, should we convert spherical -> Cartesian and then dump. That's what I think most codes that dump `wfn` do internally.

Reply:
Agreed. Idem. Will be fixed in a not-so distant future PR.